src/Bio/VCF/Parser/Parser.hs

```diff
@@ -1,22 +1,58 @@
 {-# LANGUAGE OverloadedStrings #-}
+
 module Bio.VCF.Parser.Parser where
 
 import Data.Attoparsec.ByteString
-import Data.Attoparsec.ByteString.Char8
+import qualified Data.Attoparsec.ByteString.Char8 as AC8
 import qualified Data.ByteString as B
-import Data.ByteString.Char8 (singleton)
+import qualified Data.ByteString.Char8 as BS8 (singleton, words)
+import Data.Word (Word8)
 import Bio.VCF.Internal.Types
 
+tabOrSpace :: Word8 -> Bool
+tabOrSpace c = c == 32 || c == 9
+
+isTab :: Word8 -> Bool
+isTab c = c == 9
+
+endOfLine :: Word8 -> Bool
+endOfLine c = c == 13 || c == 10
+
 parseMetaInformation :: Parser B.ByteString 
 parseMetaInformation = do
```

Comment:
This grammar looks simple enough that you could compose most of this using `Applicative` instead of `Monad` for the parsing. The compiler can parallelize some stuff with `Applicative` that it cannot parallelize with `Monad`. Also, it's good practice to use the least powerful (most general) abstraction.